From fc059adc9130acdbdedea9c055c76c44c4b5078a Mon Sep 17 00:00:00 2001
From: Geetha sowjanya <gakula@marvell.com>
Date: Mon, 1 Apr 2019 17:33:49 +0530
Subject: [PATCH 135/767] octeontx2-af: Fix rvu probe on cgx disable

commit 75c5ddc426e1116ea693ad9bfca24d63ed8b94b4 from
git@git.assembla.com:cavium/WindRiver.linux.git

RVU probe fails when no cgx device are connected.
This patch fix the issue.

Change-Id: Ied3c4a2092910457e9a2abdfde396ae129ff554a
Signed-off-by: Geetha sowjanya <gakula@marvell.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/cgx.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
index b6fbc5ce02e0..f0b1bb71164a 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
@@ -731,7 +731,7 @@ int cgx_get_mkex_prfl_info(u64 *addr, u64 *size)
 	if (!addr || !size)
 		return -EINVAL;
 
-	cgx_dev = list_first_entry(&cgx_list, struct cgx, cgx_list);
+	cgx_dev = list_first_entry_or_null(&cgx_list, struct cgx, cgx_list);
 	if (!cgx_dev)
 		return -ENXIO;
 
@@ -838,7 +838,7 @@ int cgx_get_fwdata_base(u64 *base)
 	struct cgx *cgx;
 	int err;
 
-	cgx = list_first_entry(&cgx_list, struct cgx, cgx_list);
+	cgx = list_first_entry_or_null(&cgx_list, struct cgx, cgx_list);
 	if (!cgx)
 		return -ENXIO;
 
-- 
2.31.1

